pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -71,7 +71,7 @@ func (r *ReconcileVitessKeyspace) reconcileShards(ctx context.Context, vtk *plan
 
 			status := vtk.Status.Shards[curObj.Spec.KeyRange.String()]
 			status.Cells = curObj.Status.Cells
-			status.HasMaster = curObj.Status.HasMaster
+			status.HasMaster = curObj.Status.Conditions[planetscalev2.HasMaster].CurrentStatus()
```

Comment:
You can't assume the status of your children has been filled in already. That happens asynchronously when the child controller processes it. You might read it back before that happens.

Reply:
Reverted back to old API, and fixed so we only set it if the child object status has been set (not empty string).